src/adapter/src/coord/command_handler.rs

```diff
@@ -531,15 +532,15 @@ impl<S: Append + 'static> Coordinator<S> {
     /// Handle termination of a client session.
     ///
     /// This cleans up any state in the coordinator associated with the session.
-    async fn handle_terminate(&mut self, session: &mut Session) {
-        self.clear_transaction(session).await;
+    async fn handle_terminate(&mut self, conn_id: ConnectionId) {
+        self.clear_transaction(&conn_id).await;
 
-        self.drop_temp_items(session).await;
+        self.drop_temp_items(None, &conn_id).await;
```

Comment:
The `None` here worries me. Today it happens to be ok (I spot checked the 8 places in `transact_inner` it's used, but there's no guarantee it'll be ok in the future. I think we might need to depend on having a `Session`.

Reply:
Yeah, I really don't like this. When I looked at `drop_temp_items` it only uses the `Session` object for the audit log. Though I can't figure out how to get this to work so that `SessionClient` has access to the `Session` object after termination AND `Terminate` has access to the session object.

One idea that I just thought of is that the Coord responds to a terminate command with the session back to the `SessionClient`. That way the Coord can use the `Session` while terminating and then the `SessionClient` gets it back after. Then I can undo all this connection ID stuff.